### PR TITLE
Toolset update: VS 2019 16.9 Preview 2 with Clang 11

### LINKED ISSUE #1471: provision-image.ps1: Fix Windows Driver Kit installation
In the toolset update #1451, I attempted to install the Windows Driver Kit when preparing the Virtual Machine Scale Set, needed for #1289. I wrote:

https://github.com/microsoft/STL/blob/ea156e730cf3370d0551a0532669b7de90c78915/azure-devops/provision-image.ps1#L251-L252

I didn't actually test this manually, I just verified the successful return code. On Discord, @ben-craig reports that running `wdksetup.exe` isn't blocking, so we should run:

`cmd /c start /wait wdksetup.exe /quiet /features +`

### PR DESCRIPTION
**Pull request by StephanTLavavej**
* Fix #1471 by properly waiting for the Windows Driver Kit installation to finish.
* Update `azure-pipelines.yml` to use the new pool `StlBuild-2020-12-08`.
  + @cbezault, I've created an identical pool `StlBuild-2020-12-08-1` for #1513 to use; please be aware of this when merging.
* Update `README.md` to mention Preview 2. (Its CMake and Ninja versions haven't changed, and we don't directly mention the Clang version.)
* Update `yvals_core.h` to require Clang 11. (Note that MSVC's `_MSC_VER` is remaining `1928` for the 16.9 release.)
* Remove the workaround for LLVM-37556. (We were treating it as an uncommented perma-workaround, but I remembered.)
* Remove many workarounds in `P1502R1_standard_library_header_units`.
* Remove the VSO-1225825 workaround in `<iterator>` and `<ranges>`.
  + @CaseyCarter and @miscco, please note this; I believe I have removed the workaround correctly, but it was unusually large.
* Unrelated cleanup in `<valarray>`: `__cpp_aligned_new` implies `if constexpr`.
* Update the vcpkg submodule for Boost 1.74.0. This is just their latest commit.
* In `CMakeLists.txt`, require Boost 1.74.0. (This, combined with the toolset update, implies that contributors will need to `git submodule update`, clean out the `vcpkg` submodule with `git clean -x -d -f`, and then bootstrap vcpkg and build boost-math.)
* Simplify conditional operators in `seed_seq::generate`.
  + Thanks to @AlexGuteniev for suggesting extracting `_Off` and noticing that `_Myvec[(_Kx - 1) % _Sx]` was unnecessarily complicated. Given `_Kx` in `[1, _Sx]`, then `_Kx - 1` is in `[0, _Sx - 1]`, so `% _Sx` does nothing.
* Take advantage of clang-format 11. This sets `AlignOperands: AlignAfterOperator` and `IndentCaseBlocks: true`.
* Run clang-format 11 over the entire codebase. It has better handling of conditional operators.

### CODE REVIEW

File: stl/inc/random [hunk 1]
```diff
@@ -190,8 +190,11 @@ public:
             for (_Kx = 0; _Kx < _Mx; ++_Kx) { // scramble and add any vector contributions
                 result_type _Rx1 =
                     1664525 * _Xor27(_UFirst[_Kx % _Nx] ^ _UFirst[(_Kx + _Px) % _Nx] ^ _UFirst[(_Kx - 1) % _Nx]);
-                result_type _Rx2 = static_cast<result_type>(
-                    (_Rx1 + (_Kx == 0 ? _Sx : _Kx <= _Sx ? _Kx % _Nx + _Myvec[(_Kx - 1) % _Sx] : _Kx % _Nx)) & _Mask);
+                result_type _Rx2 = static_cast<result_type>((_Rx1
+                                                                + (_Kx == 0      ? _Sx
+                                                                    : _Kx <= _Sx ? _Kx % _Nx + _Myvec[(_Kx - 1) % _Sx]
+                                                                                 : _Kx % _Nx))
+                                                            & _Mask);
```
Comment (miscco) [inline stl/inc/random:197]: I am not sure if this is really an improvement

Comment (miscco) [reply stl/inc/random:197]: Could we give the conditional expression a name?

File: stl/inc/xstring [hunk 2]
```diff
@@ -2200,10 +2200,11 @@ public:
     // length of internal buffer, [1, 16]:
     static constexpr size_type _BUF_SIZE = 16 / sizeof(value_type) < 1 ? 1 : 16 / sizeof(value_type);
     // roundup mask for allocated buffers, [0, 15]:
-    static constexpr size_type _ALLOC_MASK =
-        sizeof(value_type) <= 1
-            ? 15
-            : sizeof(value_type) <= 2 ? 7 : sizeof(value_type) <= 4 ? 3 : sizeof(value_type) <= 8 ? 1 : 0;
+    static constexpr size_type _ALLOC_MASK = sizeof(value_type) <= 1   ? 15
+                                             : sizeof(value_type) <= 2 ? 7
```
Comment (miscco) [inline stl/inc/xstring:2204]: Is there any chance to remove the additional 2 spaces before the `:`

**Review by cbezault — APPROVED** [review]
The LGTM minus the readability regression that @Miscco already pointed out in `<random>`.

File: stl/inc/xstring (hunk 2 above)
Comment (StephanTLavavej) [reply stl/inc/xstring:2204]: Not without disabling clang-format here.

File: stl/inc/random (hunk 1 above)
Comment (StephanTLavavej) [reply stl/inc/random:197]: Yeah, this one is kind of a mess. I can opt-out here.

File: stl/inc/xstring (hunk 2 above)
Comment (StephanTLavavej) [reply stl/inc/xstring:2204]: Resolving with no change since I don't see how to improve this.

Comment (StephanTLavavej) [reply stl/inc/xstring:2204]: Actually, it looks like [clang-format 11](https://releases.llvm.org/11.0.0/tools/clang/docs/ClangFormatStyleOptions.html) has a new option to deal with this, `AlignOperands: AlignAfterOperator`

File: stl/inc/random (hunk 1 above)
Comment (StephanTLavavej) [reply stl/inc/random:197]: I will totally refactor this.

**Comment by StephanTLavavej** [comment]
@CaseyCarter requested isolating the automated clang-format changes in a separate commit that `git blame` can be told to ignore, so I've rebased this PR into two commits (apologies for disrupting any review).

**Review by cbezault — APPROVED** [review]
The new changes look fine to me.

File: tests/std/tests/P1502R1_standard_library_header_units/test.cpp [hunk 3]
```diff
@@ -176,12 +176,8 @@ int main() {
         const to_chars_result result = to_chars(buf, end(buf), 3.14);
         assert(result.ec == errc{});
         assert(result.ptr == end(buf));
-#ifdef MSVC_INTERNAL_TESTING // TRANSITION, DevCom-1224512 (char_traits)
```
Comment (CaseyCarter) [inline tests/std/tests/P1502R1_standard_library_header_units/test.cpp:179]: I did a double take after reading this as `MSVC_INFERNAL_TESTING`.

**Review by CaseyCarter — APPROVED** [review]

File: tests/std/tests/P1502R1_standard_library_header_units/test.cpp (hunk 3 above)
Comment (StephanTLavavej) [reply tests/std/tests/P1502R1_standard_library_header_units/test.cpp:179]: This has definitely found a bunch of infernal compiler errors. :smiling_imp:

**Pull request merged**